src/test/java/http_server/RouterTest.java

```diff
@@ -0,0 +1,139 @@
+    HashMap<String, HashMap<String, Callback>> collection;
+    String crlf = "\r\n";
+    String request_line = "POST /echo_body HTTP/1.1" + crlf;
+    String user_agent = "User-Agent: HTTPTool/1.0" + crlf;
+    String content_type = "Content-Type: text/plain" + crlf;
+    String content_length = "Content-Length: 47" + crlf;
+    String body = crlf + "Here are all my favorite movies:\n" + "- Harry Potter";
+    String request = request_line + user_agent + content_type + content_length + body;
+    RequestParser req = new RequestParser(request);
+    Response res = new Response(req, new Router());
+
+    String anonymousFn_True_Result = "true";
+    Callback anonymousFn_True = (RequestParser req, Response res) -> {
+        res.responseBody = anonymousFn_True_Result;
+    };
+
+    String anonymousFn_False_Result = "false";
+    Callback anonymousFn_False = (RequestParser req, Response res) -> {
+        res.responseBody = anonymousFn_False_Result;
+    };
+
+    @Before
+    public void initializeRouter() {
+        router = new Router();
+        collection = router.collection();
+    }
+
+    @After
+    public void cleanUpRouter() {
+        collection.clear();
+    }
+
+    @Test
+    public void Router_Stores_Collection_Of_Methods_and_Routes() {
+        HashMap<String, HashMap<String, Callback>> testRoutesCollection = new HashMap<>();
+        Router router = new Router();
+        HashMap<String, HashMap<String, Callback>> collection = router.collection();
+
+        assertEquals(testRoutesCollection, collection);
+    }
+
+    @Test
+    public void Get_Adds_New_Get_Route_In_Collection() {
+        router.get("/get_with_body", anonymousFn_True);
+
+        assertEquals("[GET]", collection.get("/get_with_body").keySet().toString());
+    }
+
+    @Test
+    public void Calling_Callback_From_Collection_Matches_Lambda_On_Creation() {
+        router.get("/get_with_body", anonymousFn_True);
+
+        collection.get("/get_with_body").get("GET").run(req, res);
+
+        assertEquals(anonymousFn_True_Result, res.getResponseBody());
+    }
+
+    @Test
+    public void A_Method_Can_Have_Multiple_Routes() {
+        router.get("/get_with_body", anonymousFn_True);
+        router.get("/another_get", anonymousFn_True);
+        router.get("/hello", anonymousFn_True);
+
+        assertEquals("[/get_with_body, /another_get, /hello]", collection.keySet().toString());
+    }
+
+//    @Test
+//    public void Each_Route_In_a_Method_Has_Its_Own_Callback() {
+//        router.get("/get_with_body", anonymousFn_True);
+//        router.get("/hello", anonymousFn_False);
+//
+//        String get_with_body_callback_result = collection.get("/get_with_body").get("GET").run(req, res);
+//        String hello_callback_result = collection.get("/hello").get("GET").run(req, res);
+//
+//        assertEquals(anonymousFn_True_Result, get_with_body_callback_result);
+//        assertEquals(anonymousFn_False_Result, hello_callback_result);
+//    }
```

Comment:
Do you need these tests if they're commented out?

If they remind you to do something, could you just leave yourself a note?

Reply:
These tests were initially used to make sure the callbacks were working as expected (diff routes have different callbacks and the callbacks themselves were executing correctly).  

But after I adjusted the callback to return `void`, these tests don't work anymore as they previously relied on returning an expected value. I am not sure if I should figure out a way to still test the callback functionality, or just remove them all together